homeassistant/components/unifiprotect/camera.py

```diff
@@ -0,0 +1,169 @@
+"""Support for Ubiquiti's UniFi Protect NVR."""
+from __future__ import annotations
+
+from collections.abc import Callable, Generator, Sequence
+import logging
+from typing import Any
+
+from pyunifiprotect.api import ProtectApiClient
+from pyunifiprotect.data import Camera as UFPCamera
+from pyunifiprotect.data.devices import CameraChannel
+
+from homeassistant.components.camera import SUPPORT_STREAM, Camera
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    ATTR_BITRATE,
+    ATTR_CHANNEL_ID,
+    ATTR_FPS,
+    ATTR_HEIGHT,
+    ATTR_WIDTH,
+    DOMAIN,
+)
+from .data import ProtectData
+from .entity import ProtectDeviceEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_camera_channels(
+    protect: ProtectApiClient,
+) -> Generator[tuple[UFPCamera, CameraChannel, bool], None, None]:
+    """Get all the camera channels."""
+    for camera in protect.bootstrap.cameras.values():
+        if len(camera.channels) == 0:
```

Comment:
```py
if not camera.channels:
```

Reply:
Note: going to resolve these as I make the changes locally for my own sanity (just so you do not think I am ignoring you).